spring-cloud-dataflow-composed-task-runner/src/main/java/org/springframework/cloud/dataflow/composedtaskrunner/ComposedRunnerJobFactory.java

```diff
@@ -112,12 +109,7 @@ public Job getObject() throws Exception {
 						.start(createFlow())
 						.end())
 				.end();
-		if(this.incrementInstanceEnabled && !this.composedTaskProperties.isUuidInstanceEnabled()) {
-			builder.incrementer(new RunIdIncrementer());
-		}
-		else if(this.composedTaskProperties.isUuidInstanceEnabled()) {
-			builder.incrementer(new UuidIncrementer());
-		}
+		builder.incrementer(new UuidIncrementer());
```

Comment:
Shouldn't this be guarded by `this.composedTaskProperties.isUuidInstanceEnabled()` ?

Reply:
Absolutely.   And the default of the isUuidInstanceEnabled needs to be set to true.   
GREAT CATCH!